video-library/videoLibrary/src/controllers/GroupController.ts

```diff
@@ -0,0 +1,151 @@
+import { Request, Response } from "express";
+import { getRepository } from "typeorm";
+import { Group } from "../entity/Group";
+
+class GroupController {
```

Comment:
budu komentit postupně.. aktualně se nedívám co máš uvnitř té classy.. to budu komentit v kroku č.2...

Tvůj cíl je, zavést nějakou design patternu která to pomůže a bude ti usnadňovat vytváření dalších class...

na to nám slouží patterna která vychází z Abstraktních tříd https://khalilstemmler.com/blogs/typescript/abstract-class/

Cílem je tedy udělat abstraktní controller kterým budeš extendovat potom tyto tvé hlavní kontrollery... tedy budeš mít BaseController který ponese třídy a attributy které budeš používat a případně overridovat

tedy například tento controller bude mít v construktoru na vstupu Entitu kterou si uloží do this.entity a udělá si this.repository které budeš používat v následujících metodách
tyto metody následně budou třeba
findById
create
update
list(s paginací)

všechny tyto metody budou abstraktní a generický.. tedy budeš to psát za pomocí generiky a na výstupu po extendu a předání Entity v super() v konstruktoru extendované classy, budeš mít otypované tyto metody a budeš moci krásně i v této extendované třídě šahat na this.entity a this.repository... tohle ti dovolí krásně genericky znovupoužívat base bussiness logiku